extensions/lifecycle/tasks/LifecycleResetTransitionInProgressTask.js

```diff
@@ -1,169 +1,28 @@
-                            eTag,
-                            rest.try,
-                            log,
-                            (err, res) => {
-                                if (err) {
-                                    return nextObject(err);
-                                }
-
-                                return nextObject(null, res + objsPerBucket);
-                            }
-                        ),
-                    (err, res) => {
-                        if (err) {
-                            return next(err);
-                        }
-
-                        return next(null, res);
-                    }
-                ),
-            (err, res) => {
-                if (err) {
-                    return cb(err);
-                }
-
-                return cb(null, res);
-            }
-        );
+        super(proc);
+        this.processName = 'transition';
     }
 
-    requeueObjectVersion(accountId, bucketName, objectKey, objectVersion, etag, try_, bucketLogger, cb) {
-        const client = this.getBackbeatMetadataProxy(accountId);
-        if (!client) {
-            return cb(errors.InternalError.customizeDescription(
-                `Unable to obtain client for account ${accountId}`,
-            ));
-        }
-
-        const params = {
-            bucket: bucketName,
-            objectKey,
-        };
-        if (objectVersion) {
-            params.versionId = objectVersion;
+    updateObjectMD(md, try_, log, etag) {
+        if (this.shouldSkipObject(md, etag, log)) {
+            return false;
         }
-
-        const log = this.logger.newRequestLogger(bucketLogger.getUids());
-        log.addDefaultFields({
-            accountId,
-            bucketName,
-            objectKey,
-            objectVersion,
-            etag,
-            try: try_,
-        });
-
-        return client.getMetadata(params, log, (err, blob) => {
-            LifecycleMetrics.onS3Request(log, 'getMetadata', 'transition', err);
-            if (err) {
-                return cb(err);
-            }
-
-            const { result: md, error } = ObjectMD.createFromBlob(blob.Body);
-            if (error) {
-                return cb(error);
-            }
-
-            if (this.shouldSkipObject(md, etag, log)) {
-                return cb(null, 0);
-            }
-
-            md.setTransitionInProgress(false);
-            md.setUserMetadata({
+        md.setTransitionInProgress(false);
+        md.setUserMetadata({
                 'x-amz-meta-scal-s3-transition-attempt': try_,
-            });
-
-            return client.putMetadata({ ...params, mdBlob: md.getSerialized() }, log,
-                err => {
-                    LifecycleMetrics.onS3Request(log, 'putMetadata', 'transition', err);
-                    if (err) {
-                        return cb(err);
-                    }
-
-                    return cb(null, 1);
-                }
-            );
         });
```

Comment:
I think there was an issue in the original bug : the "originOp" is not set when writing the metadata, which may cause extra notification... To avoid any issue, we should probably set it now (`setOriginOp("s3:LifecycleTransition:Retry")`) to avoid issues...

(it can be done in a separate dedicated ticket)

Reply:
Created a ticket for that: https://scality.atlassian.net/browse/BB-421